platform/darwin/src/MGLBaseStyleLayer.h

```diff
@@ -0,0 +1,26 @@
+#import <Foundation/Foundation.h>
+
+@interface MGLBaseStyleLayer : NSObject
+
+@property (nonatomic, assign, getter=isVisible) BOOL visible;
+
+/**
+ The maximum zoom level on which the layer gets parsed and appears on.
+ */
+@property (nonatomic, assign) float maxZoom;
+
+/**
+ The minimum zoom level on which the layer gets parsed and appears on.
+ */
+@property (nonatomic, assign) float minZoom;
+
+/**
+ Updates the layers layout and paint properties.
```

Comment:
I think this method should be moved to `MGLStyle`. You call `-update` on the layer but it will actually recalculate styles and update classes for all layers.